android/console.c

```diff
@@ -1817,6 +1818,35 @@ do_gsm_report( ControlClient  client, char*  args )
     return 0;
 }
 
+static int
+do_gsm_timezone( ControlClient  client, char  *args)
+{
+    if (args) {
+        if (!strcmp(args, "auto")) {
+            // reset timezone
+            amodem_reset_timezone(client->modem);
+        } else {
+            // check argument format
+            int i = (args[0] == '+' || args[0] == '-') ? 1 : 0;
+            int length = strlen(args);
+            while (i < length) {
+                if (!isdigit(args[i])) {
+                    control_write( client, "KO: tzdiff is neither a number nor 'auto'\r\n");
+                    return -1;
+                }
+                i++;
+            }
+
+            // update timezone
+            int tzdiff = atoi(args);
+            amodem_set_timezone(client->modem, tzdiff);
+        }
+    }
+
+    control_write( client, "tzdiff = %d\r\n", amodem_get_timezone(client->modem));
```

Comment:
Nit: Just print the result without 'tzdiff = ', it makes the command user parses the result easier.